packages/fleather/lib/src/widgets/text_line.dart

```diff
@@ -166,8 +169,10 @@ class _TextLineState extends State<TextLine> {
 
   InlineSpan _segmentToTextSpan(Node segment, FleatherThemeData theme) {
     if (segment is EmbedNode) {
-      return WidgetSpan(
-          child: EmbedProxy(child: widget.embedBuilder(context, segment)));
+      // TODO: Why do we need EmbedProxy? Things work fine without it.
+      // return WidgetSpan(
+      //     child: EmbedProxy(child: widget.embedBuilder(context, segment)));
+      return WidgetSpan(child: widget.embedBuilder(context, segment));
```

Comment:
I couldn't notice the difference here with or without `EmbedProxy`.

Reply:
EmbedProxy is useful for selection. is it the reason you cannot move without it?

Reply:
Play with selecting the content and check the cursor size also around your image with/without.

Reply:
To me it works the same with and without `EmbedProxy` :shrug: 
I have reverted the change anyway but something you might play with.